application/config/config.go

```diff
@@ -213,14 +214,22 @@ func New() *Config {
 	c.deviceId = c.determineDeviceId()
 	c.addDefaults()
 	c.filterSeverity = lsp.DefaultSeverityFilter()
-	c.engine = app.CreateAppEngine()
+	initWorkFlowEngine(c)
 	err := c.engine.Init()
 	if err != nil {
 		log.Warn().Err(err).Msg("Failed to initialize workflow engine")
 	}
 	return c
 }
 
+func initWorkFlowEngine(c *Config) {
```

Comment:
Should initialisation logic leave in configuration? I suggest we keep this out of config to keep SRP.

Reply:
In the future this should done with dependency injection and initialization. right now, I don't think we can do it without cyclical deps between di.Init and config.

Reply:
Can we leave a comment for that here please? We can also move this to a separate file for at least some visual separation of concerns.